zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/BerryClient.scala

```diff
@@ -150,7 +154,11 @@ class BerryClient(val jsonParser: JSONParser,
     val futures = Future.traverse(queryGroup.queries) { queryInfo =>
       if (queryInfo.queryBound.overlaps(interval)) {
         val overlaps = queryInfo.queryBound.overlap(interval)
-        val timeFilter = FilterStatement(queryInfo.dataSetInfo.schema.getTimeField.get, None, Relation.inRange,
+        if (!queryInfo.dataSetInfo.schema.isInstanceOf[Schema]){
+          ???
```

Comment:
if we have previously checked it already, we should not do it again? (especially for the private function.)